org-static-blog.el

```diff
@@ -445,40 +455,51 @@ e.g. `(('foo' 'file1.org' 'file2.org') ('bar' 'file2.org'))`"
             (push (cons tag (list post-filename)) tag-tree)))))
     tag-tree))
 
+(defun org-static-blog--preview-region ()
+  "Find the start and end of the preview in the current buffer."
+  (goto-char (point-min))
+  (if org-static-blog-preview-end
+      (when (or (search-forward (or org-static-blog-preview-start "<p>") nil t)
+                (search-forward "<p>" nil t))
+        (let ((start (match-beginning 0)))
+          (or (search-forward org-static-blog-preview-end nil t)
+              (search-forward "</p>" nil t))
+          (cons start (point))))
+    (when (search-forward (or org-static-blog-preview-start "<p>") nil t)
+      (let ((start (match-beginning 0)))
+        (search-forward "</p>")
+        (cons start (point))))))
+
 (defun org-static-blog-get-preview (post-filename)
   "Get title, date, tags from POST-FILENAME and get the first paragraph from the rendered HTML.
 If the HTML body contains multiple paragraphs, include only the first paragraph,
 and display an ellipsis.
 Preamble and Postamble are excluded, too."
   (with-temp-buffer
     (insert-file-contents (org-static-blog-matching-publish-filename post-filename))
-    (let ((post-title)
-          (post-date)
-          (post-taglist)
-          (post-ellipsis "")
-          (first-paragraph-start)
-          (first-paragraph-end))
-      (setq post-title (org-static-blog-get-title post-filename))
-      (setq post-date (org-static-blog-get-date post-filename))
-      (setq post-taglist (org-static-blog-post-taglist post-filename))
-      ;; Find where the first paragraph ends and starts
-      (goto-char (point-min))
-      (when (search-forward "<p>" nil t)
-        (search-forward "</p>")
-        (setq first-paragraph-end (point))
-        (search-backward "<p>")
-        (setq first-paragraph-start (point))
-        (goto-char first-paragraph-end)
-        (when (search-forward "<p>" nil t)
-          (setq post-ellipsis (concat (when org-static-blog-preview-link-p
-                                        (format "<a href=\"%s\">" (org-static-blog-get-post-url post-filename)))
-                                      org-static-blog-preview-ellipsis
-                                      (when org-static-blog-preview-link-p "</a>\n")))))
+    (let* ((post-title (org-static-blog-get-title post-filename))
+           (post-date (org-static-blog-get-date post-filename))
+           (post-taglist (org-static-blog-post-taglist post-filename))
+           (post-ellipsis "")
+           (prev-region (org-static-blog--preview-region))
```

Comment:
If `org-static-blog--preview-region` returned a string instead of two positions, this `let*` could be reverted to a straight `let`, and the code would simplify a bit.